File: snippets/php/check_the_status_of_a_batch_operation.php
require_once('/path/to/MailchimpMarketing/vendor/autoload.php');

$mailchimp = new MailchimpMarketing\ApiClient();

$mailchimp->setConfig([
    'apiKey' => 'YOUR_API_KEY',
    'server' => 'YOUR_SERVER_PREFIX'
]);

$batch_id = 'YOUR_BATCH_ID';

try {
    $response = $mailchimp->batches->status($batch_id);
    echo $response["status"];
} catch (\MailchimpMarketing\ApiException $e) {
    echo $e->getMessage();
}

File: snippets/ruby/check_the_status_of_a_batch_operation.rb
require "MailchimpMarketing"

mailchimp = MailchimpMarketing::Client.new
mailchimp.set_config({
  :api_key => ENV["API_KEY"],
  :server => ENV["API_SERVER"]
})
batch_id = ENV["BATCH_OPERATION_ID"]

response = mailchimp.batches.status batch_id
puts response.status